tasks/task_manager.cc

```diff
@@ -117,48 +179,79 @@ bool task_manager::task::impl::is_done() const noexcept {
     return _status.state == tasks::task_manager::task_state::done;
 }
 
+future<std::vector<task_manager::task::task_essentials>> task_manager::task::impl::get_failed_children() const {
```

Comment:
why does this function need to be async?

Reply:
where? maybe I missed it but it seems completely synchronous, no?

Reply:
`map_each_task` is async as it holds a lock